libs/aws/langchain_aws/retrievers/bedrock.py

```diff
@@ -128,6 +170,7 @@ def _get_relevant_documents(
         response = self.client.retrieve(
             retrievalQuery={"text": query.strip()},
             knowledgeBaseId=self.knowledge_base_id,
+            nextToken=None,
```

Comment:
Should this be handled to extract the next set of results, or parameterized from `kwargs` so that the users can send the token and get the next set of documents?

Reply:
To implement the handling of `next_token`, I had to:

-  create another method called `_get_relevant_documents_with_token` that will return the list of `Documents` and the token as a tuple.
- This new method was created because the  method `_get_relevant_documents` from the class `BaseRetriever` from which this class inherits has a signature of `List[Document]`.
- Remove the `token` usage from `_get_relevant_documents` because of the duplication with the previous method.